File: CC_C#/BVCS Projects/Times/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Times
{
    partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        int product;
        int numProb;
        int numRight;
        Random myRandom = new Random();

        private void Form1_Load(object sender, EventArgs e)
        {
            // Initialize variables
            numProb = 0;
            numRight = 0;
            // display the first problem
            btnNext.PerformClick();
        }

        private void btnNext_Click(object sender, EventArgs e)
        {
            // Generate next multiplication problem
            int number1, number2;
            txtAnswer.Text = "";
            lblMessage.Text = "";
            numProb++;
            // Generate random numbers for factors
            number1 = myRandom.Next(10);
            number2 = myRandom.Next(10);
            lblNum1.Text = Convert.ToString(number1);
            lblNum2.Text = Convert.ToString(number2);
            // Find product
            product = number1 * number2;
            btnNext.Enabled = false;
            txtAnswer.Focus();
        }

        private void txtAnswer_KeyPress(object sender, KeyPressEventArgs e)
        {
            int ans;
            // Check for number only input and for return key
            if ((e.KeyChar >= '0' && e.KeyChar <= '9') || (int) e.KeyChar == 8)
            {
                e.Handled = false;
            }
            else if ((int) e.KeyChar == 13)
            {
                // Check answer and update score
                ans = Convert.ToInt32(txtAnswer.Text);
                if (ans == product)
                {
                    numRight++;
                    lblMessage.Text = "That's correct!";
                }
                else
                {
                    lblMessage.Text = "Answer is " + Convert.ToString(product);
                }
                lblScore.Text = String.Format("{0:f0}", 100 * ((double) numRight / numProb)) + "%";
                btnNext.Enabled = true;
                btnNext.Focus();
            }
            else
            {
                e.Handled = true;
            }
        }
    }
}

File: CC_Visual Basic/VisualBasic/BeginVBE/BeginVBE/BVBE Projects/Times/Form1.vb
Public Class Form1
  Dim Product As Integer
  Dim NumProb As Integer
  Dim NumRight As Integer
  Dim MyRandom As New Random
  Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
    'Initialize variables
    NumProb = 0
    NumRight = 0
    'display the first problem
    btnNext.PerformClick()
  End Sub
  Private Sub btnNext_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnNext.Click
    'Generate next multiplication problem
    Dim Number1 As Integer
    Dim Number2 As Integer
    txtAnswer.Text = ""
    lblMessage.Text = ""
    NumProb = NumProb + 1
    'Generate random numbers for factors
    Number1 = MyRandom.Next(10)
    Number2 = MyRandom.Next(10)
    lblNum1.Text = Format(Number1, "0")
    lblNum2.Text = Format(Number2, "0")
    'Find product
    Product = Number1 * Number2
    btnNext.Enabled = False
    txtAnswer.Focus()
  End Sub
  Private Sub txtAnswer_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtAnswer.KeyPress
    Dim Ans As Integer
    'Check for number only input and for return key
    If (e.KeyChar >= "0" And e.KeyChar <= "9") Or e.KeyChar = ControlChars.Back Then
      e.Handled = False
    ElseIf e.KeyChar = ControlChars.Cr Then
      'Check answer and update score
      Ans = Val(txtAnswer.Text)
      If Ans = Product Then
        NumRight = NumRight + 1
        lblMessage.Text = "That's correct!"
      Else
        lblMessage.Text = "Answer is " + Format(Product, "0")
      End If
      lblScore.Text = Format(100 * (NumRight / NumProb), "0") + "%"
      btnNext.Enabled = True
      btnNext.Focus()
    Else
      e.Handled = True
    End If
  End Sub

End Class
